lib/model/model.go

```diff
@@ -253,24 +255,44 @@ func NewModel(cfg config.Wrapper, id protocol.DeviceID, clientName, clientVersio
 }
 
 func (m *model) serve(ctx context.Context) error {
-	// Add and start folders
+	defer m.closeAllConnectionsAndWait()
+
+	if err := m.initFolders(); err != nil {
+		close(m.started)
+		return fatalErr(err)
+	}
+
+	m.cfg.Subscribe(m)
```

Comment:
We could miss a config change after initFolders but before we subscribe, so perhaps subscribe before initFolders?

Reply:
Probably better that way round, yes. However really there should be a hook or something on `Subscribe` to initialize the subscriber with the config that's in place while subscribing. Otherwise you can still have the race that a config change comes in while we are still initialising folders (which isn't even so unlikely, given creating the filesets and recalculating metadata can take a while).